src/core/network/qgsfiledownloader.h

```diff
@@ -108,7 +108,7 @@ class CORE_EXPORT QgsFileDownloader : public QObject
     void onSslErrors( QNetworkReply *reply, const QList<QSslError> &errors );
 #endif
 
-  protected:
+  public:
```

Comment:
is this required?

Reply:
I do think it is required, making the desctructor protected in this case will make it impossible to allocate QgsFileDownloader as a local variable and also prevent it from being used in unique_ptr. in the other places where QgsFileDownloader is used, there appears to be a memory leak also. I think we should change it to public.